### ISSUE TITLE
Additional loads ending up in the wrong pool

### ISSUE THREAD

**Issue by philrz** [brimdata/brim#2599]
Repro is with Brim commit 643c6a8.

Repro steps as shown in the attached video:

1. Drag the [sample.zng](https://github.com/brimdata/brim/blob/main/src/test/shared/data/sample.zng) test data into the app twice, creating two separate pools
2. Click on the first pool and re-drag the same file as additional data
3. Click on the second pool and re-drag the same file as additional data
4. Query the `count()` for each pool

Given the test data file contains 31 records, we'd expect each pool to have a count of 62 records at the end. Instead we find that the first pool has just the original 31 records, whereas both the additional loads ended up in the second pool.

https://user-images.githubusercontent.com/5934157/203452443-f8090af8-cbdc-4cb0-be7d-6781f64e1f25.mp4

**Comment by jameskerr** [comment]
Oh no!

**Comment by philrz** [comment]
I happened to revisit this bug and found it seemed to have been fixed. I did a binary search and confirmed that it got fixed with the changes in #2601. The attached video verifies this by repeating the original repro steps at the commit associated with the PR. As it shows, the count of records shows the same value for each pool as we would expect.

https://user-images.githubusercontent.com/5934157/214678992-6d6d5a6e-7b45-44d5-ab38-a1761773599a.mp4

Thanks @jameskerr!
